### PR TITLE
Tidy telecomms radio code, make PDA server real telecomms machinery

### LINKED ISSUE #32148: Syndicate Cyborgs' radio messages get duplicated.
The first message gets sent instantly as normal and a copy of the message gets sent a few seconds later.

Round ID: 77990

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
refactor: Telecomms and radio code has been cleaned up.
tweak: The PDA message server is now a real piece of telecomms machinery.
tweak: PDA logs now include the job as well as name of the participants.
fix: The Syndicate frequency no longer hears its own messages twice.
fix: Talking directly into a radio which is also hot-miked no longer double-talks.
/:cl:

Follow-up to #33381 and last big change in this series. Radio broadcasting code is redone. The PDA server is made a real piece of telecomms machinery (PDA messages previously already faked subspace transmissions to check if telecomms was up, now it's real). The PDA message monitoring computer has also received a couple of tweaks (like fixing the broken `<A>` tags).

This has been tested in several real rounds privately with no noticeable ill effects, but it definitely should be testmerged before being merged.

I considered splitting this into a PR for the radio code and a PR for the PDA server, but it would involve hacking old PDAs to work on the new broadcasting and then immediately deleting that code, so I decided against it for now.

Fixes #32148.

### CODE REVIEW

File: code/game/machinery/telecomms/broadcasting.dm [hunk 1]
```diff
@@ -53,127 +53,141 @@
...
-						var/message, var/name, var/job, var/realname,
-						var/data, var/compression, var/list/level, var/freq, var/list/spans,
-						var/verb_say, var/verb_ask, var/verb_exclaim, var/verb_yell, var/datum/language/language)
-
+/datum/signal/subspace/proc/broadcast()
 	set waitfor = FALSE
 
-	message = copytext(message, 1, MAX_BROADCAST_LEN)
+// Vocal transmissions (i.e. using saycode).
+// Despite "subspace" in the name, these transmissions can also be RADIO
```
Comment (AnturK) [inline code/game/machinery/telecomms/broadcasting.dm]: Why the need for the /subpsace subtype here then ? Seems like modifying the basic datum would make more sense ?
As in add the common functionality to basic signal then just have signal/vocal ? original message/server and levels seem pretty generic properties to me.

File: code/game/machinery/telecomms/broadcasting.dm [hunk 2]
```diff
@@ -53,127 +53,141 @@
...
-
-  //#### Sending the signal to all subspace receivers ####//
-	for(var/obj/machinery/telecomms/receiver/R in GLOB.telecomms_list)
-		R.receive_signal(signal)
-
-	sleep(rand(10,25))
+		SSblackbox.LogBroadcast(frequency)
 
-	return signal
+	QDEL_IN(virt, 50)  // Make extra sure the virtualspeaker gets qdeleted
```
Comment (AnturK) [inline code/game/machinery/telecomms/broadcasting.dm]: Do signals actually stay around after use for this to fire ?

File: code/game/machinery/telecomms/machines/broadcaster.dm [hunk 3]
```diff
@@ -16,70 +16,45 @@ GLOBAL_VAR_INIT(message_delay, 0) // To make sure restarting the recentmessages
...
-							  signal.data["realname"], 4, signal.data["compression"], signal.data["level"], signal.frequency, signal.data["spans"],
-							  signal.data["verb_say"], signal.data["verb_ask"], signal.data["verb_exclaim"], signal.data["verb_yell"], signal.data["language"])
+	signal.broadcast()
 
-		if(!GLOB.message_delay)
-			GLOB.message_delay = 1
-			spawn(10)
-				GLOB.message_delay = 0
-				GLOB.recentmessages = list()
+	if(!GLOB.message_delay)
```
Comment (AnturK) [inline code/game/machinery/telecomms/machines/broadcaster.dm]: This var needs another name, not your fault though.

File: code/game/machinery/telecomms/machines/message_server.dm [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+/datum/signal/subspace/pda/Topic(href, href_list)
+	..()
+	if (href_list["photo"])
+		var/mob/M = usr
+		M << browse_rsc(data["photo"], "pda_photo.png")
+		M << browse("<html><head><title>PDA Photo</title></head>" \
+		+ "<body style='overflow:hidden;margin:0;text-align:center'>" \
+		+ "<img src='pda_photo.png' width='192' style='-ms-interpolation-mode:nearest-neighbor' />" \
+		+ "</body></html>", "window=book;size=192x192")
```
Comment (AnturK) [inline code/game/machinery/telecomms/machines/message_server.dm]: `window=pdaphoto` and `onclose(M,"pdaphoto")` or similar. (This is probably my copypaste 🙈 )

File: code/game/machinery/telecomms/machines/message_server.dm [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+/datum/data_pda_msg/Topic(href,href_list)
+	..()
+	if(href_list["photo"])
+		var/mob/M = usr
+		M << browse_rsc(photo, "pda_photo.png")
+		M << browse("<html><head><title>PDA Photo</title></head>" \
+		+ "<body style='overflow:hidden;margin:0;text-align:center'>" \
+		+ "<img src='pda_photo.png' width='192' style='-ms-interpolation-mode:nearest-neighbor' />" \
+		+ "</body></html>", "window=book;size=192x192")
```
Comment (AnturK) [inline code/game/machinery/telecomms/machines/message_server.dm]: Same thing here.

**Review by AnturK — COMMENTED** [review]
Good stuff.

File: code/game/machinery/telecomms/broadcasting.dm (hunk 2 above)
Comment (SpaceManiac) [reply code/game/machinery/telecomms/broadcasting.dm]: The signal doesn't need to stay around, `QDEL_IN` sets a global timer:
```dm
#define QDEL_IN(item, time) addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, item), time, TIMER_STOPPABLE)
```

**Pull request merged**